plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/simulate/SimulatedItemResponse.java

```diff
@@ -0,0 +1,171 @@
+import org.elasticsearch.ingest.Data;
+import org.elasticsearch.rest.RestStatus;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class SimulatedItemResponse implements Streamable, StatusToXContent {
+
+    private Data data;
+    private List<ProcessedData> processedDataList;
+    private Throwable failure;
+
+    public SimulatedItemResponse() {
+
+    }
+
+    public SimulatedItemResponse(Data data) {
+        this.data = data;
+    }
+
+    public SimulatedItemResponse(List<ProcessedData> processedDataList) {
+        this.processedDataList = processedDataList;
+    }
+
+    public SimulatedItemResponse(Throwable failure) {
+        this.failure = failure;
+    }
+
+    public boolean isFailed() {
+        return this.failure != null;
+    }
+
+    public boolean isVerbose() {
+        return this.processedDataList != null;
+    }
+
+    public Data getData() {
+        return data;
+    }
+
+    public List<ProcessedData> getProcessedDataList() {
+        return processedDataList;
+    }
+
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        boolean isFailed = in.readBoolean();
+        boolean isVerbose = in.readBoolean();
+        if (isFailed) {
+            this.failure = in.readThrowable();
+            // TODO(talevy): check out mget for throwable limitations
+        } else if (isVerbose) {
+            int size = in.readVInt();
+            processedDataList = new ArrayList<>();
+            for (int i = 0; i < size; i++) {
+                ProcessedData processedData = new ProcessedData();
+                processedData.readFrom(in);
+                processedDataList.add(processedData);
+            }
+        } else {
+            String index = in.readString();
+            String type = in.readString();
+            String id = in.readString();
+            Map<String, Object> doc = in.readMap();
+            this.data = new Data(index, type, id, doc);
+        }
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeBoolean(isFailed());
+        out.writeBoolean(isVerbose());
+
+        if (isFailed()) {
+            out.writeThrowable(failure);
+        } else if (isVerbose()) {
+            out.writeVInt(processedDataList.size());
+            for (ProcessedData p : processedDataList) {
+                p.writeTo(out);
+            }
+        } else {
+            out.writeString(data.getIndex());
+            out.writeString(data.getType());
+            out.writeString(data.getId());
+            out.writeMap(data.getDocument());
+        }
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        builder.field(Fields.ERROR, isFailed());
+        builder.field(Fields.VERBOSE, isVerbose());
```

Comment:
verbosity level should be shared by `SimulatePipelineResponse` one time, not at each doc response